hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -368,16 +377,23 @@ public CleanPlanner(HoodieEngineContext context, HoodieTable<T, I, K, O> hoodieT
   
   private List<CleanFileInfo> getReplacedFilesEligibleToClean(List<String> savepointedFiles, String partitionPath, Option<HoodieInstant> earliestCommitToRetain) {
     final Stream<HoodieFileGroup> replacedGroups;
-    if (earliestCommitToRetain.isPresent()) {
+    if (earliestCommitToRetain.isPresent() && !isDeletePartitionOperation(earliestCommitToRetain)) {
```

Comment:
not sure if we can do this. 
lets say, 
C1, C2... C4(clustering in partition P1),... C6(delete partition with partition P2)...

and earliest commit to retain is 
1. lets say C6. we need to delete all file groups in P2 and any other replaced file groups < C6. i.e. fileSystemView.getReplacedFileGroupsBefore(earliestCommitToRetain.get().getTimestamp(), partitionPath);
2. lets say C7. again, we need to delete all file groups in P2 and any other replaced file groups < C7. 

Can you check the logic if we are covering both these cases.